sonic_platform_base/sonic_xcvr/api/public/cmis.py

```diff
@@ -180,24 +180,24 @@ def get_transceiver_bulk_status(self):
         bulk_status = {
             "rx_los": all(rx_los.values()) if self.get_rx_los_support() else 'N/A',
             "tx_fault": all(tx_fault.values()) if self.get_tx_fault_support() else 'N/A',
-            "tx_disable": all(tx_disable),
+            "tx_disable": all(tx_disable) if self.get_tx_disable_support() else 'N/A',
             "tx_disabled_channel": tx_disabled_channel,
             "temperature": temp,
             "voltage": voltage
         }
 
         for i in range(1, self.NUM_CHANNELS + 1):
-            bulk_status["tx%dbias" % i] = tx_bias['LaserBiasTx%dField' % i]
-            bulk_status["rx%dpower" % i] = rx_power['OpticalPowerRx%dField' %i]
-            bulk_status["tx%dpower" % i] = tx_power['OpticalPowerTx%dField' %i]
+            bulk_status["tx%dbias" % i] = tx_bias['LaserBiasTx%dField' % i] if self.get_tx_bias_support() else 'N/A'
```

Comment:
There are two things we need to check:
1. Module is flat memory module
2. Module is non-flat memory module but still doesn't support tx bias, rx power, bias current monitoring (See page 01, byte 160)

Please modify the get_tx_bias_support as:
```
def get_tx_bias_support():
   if is not self.flat_memory() and page1, byte 160, bit 0 is set:
        return True
    return false
```

Reply:
done

Reply:
get_tx_bias_support() is already present inside get_tx_bias() so this extra if check is not required here.

Reply:
it will return a list of 'N/A' if not supported. But tx_bias['LaserBiasTx%dField' % i] will return KeyError. That's why I add this if else statement in this line.